agenda/data/src/main/java/com/tasky/agenda/data/repositories/OfflineFirstEventRepository.kt

```diff
@@ -89,7 +91,22 @@ class OfflineFirstEventRepository(
     }
 
     override suspend fun deleteLocalAttendeeFromEvent(eventId: String): EmptyDataResult<DataError.Network> {
-        return remoteEventDataSource.deleteLocalAttendeeFromAnEvent(eventId)
+        val result = remoteEventDataSource.deleteLocalAttendeeFromAnEvent(eventId)
+        if (result is Result.Error) {
+            val event = localEventDataSource.getEventById(eventId) ?: return result
+            val userId = authInfoStorage.get()?.userId ?: return result
+            applicationScope.launch {
+                eventSyncScheduler.sync(EventSyncScheduler.SyncType.UpdateEventSync(
+                    event.copy(
+                        attendees = event.attendees.filter {
+                            it.userId != userId || event.host == userId
+                        }
+                    )
+                ))
+            }.join()
+        }
+        localEventDataSource.deleteEvent(eventId)
```

Comment:
If you’ve successfully performed the sync call, then we want to guarantee that the local is also deleted. `applicationScope` can be used to safeguard against premature cancellation of our coroutines.

Also, would this flow not make more sense for an offline-first app?
1. Delete the event app locally
2. If successful, attempt to delete the event remotely
3. If API call failed, then schedule a sync call

Reply:
Yes